scripts/commands/gotoid.lua

```diff
@@ -0,0 +1,71 @@
+---------------------------------------------------------------------------------------------------
+-- func: gotoid
+-- desc: Go to given mob or npc ID
+---------------------------------------------------------------------------------------------------
+require("scripts/globals/status")
+
+cmdprops =
+{
+    permission = 1,
+    parameters = "i"
+}
+
+function error(player, msg)
+    player:PrintToPlayer(msg)
+    player:PrintToPlayer("!gotoid <mobId|npcId>")
+end
+
+function onTrigger(player, target)
+
+    -- validate npc
+    if not target or target == 0 then
+        error(player, "You must enter a mob or NPC ID.")
+        return
+    end
+
+    -- is entity up?
+    local isUp = false
+    local targ = GetNPCByID(target)
+    local pos0 = false
+    
+    if targ then
+        local pos = targ:getPos()
+        pos0 = (pos.x == 0 and pos.y == 0 and pos.z == 0)
+        if targ:getStatus() == dsp.status.NORMAL and not pos0 then
+            isUp = true
+        end
+    else
+        targ = GetMobByID(target)
+        if targ then
+            local pos = targ:getPos()
+            pos0 = (pos.x == 0 and pos.y == 0 and pos.z == 0)
+            if targ:isSpawned() and not pos0 then
+                isUp = true
+            end
+        end
+    end
+    
+    if not targ then
+        player:goToEntity(target)
+    elseif pos0 then
+        player:PrintToPlayer(string.format("%s (%i) has not been given coordinates.", targ:getName(), targ:getID()))
+    else
+        -- determine whether we need zoneId parameter
+        local gotoZone = nil
+        if targ:getZoneID() ~= player:getZoneID() then
+            gotoZone = targ:getZoneID()
+        end
+
+        -- display message
+        if isUp then
+            player:PrintToPlayer(string.format("Going to %s (%i).", targ:getName(), targ:getID()))
+        else
+            player:PrintToPlayer(string.format("%s (%i) is not currently up. Going to last known coordinates.", targ:getName(), targ:getID()))
+        end
+
+        -- half a second later, go.  this delay gives time for previous message to appear
+        player:timer(500, function(player)
+            player:setPos(targ:getXPos(), targ:getYPos(), targ:getZPos(), targ:getRotPos(), gotoZone)
```

Comment:
`player:goToEntity(target)` actually initiates the position change for you core side already
https://github.com/DarkstarProject/darkstar/blob/09b3ddfe5f24190203a2ceb225c479bcc699d045/src/map/message.cpp#L439

Reply:
target is the ID (it's the number passed in by the player).  targ is the entity.

Reply:
Ok. Just need to rework your message placement and kill the setpos then

Reply:
the messages and setpos are there purposefully.  If it can't locate the entity ID on current map server, it uses goToEntity.  If it finds them on the current map server, it setPos'es you to the x/y/z or the x/y/z/zone if in different zone.  If the mob/npc was spawned but currently isn't, it sends you to last known coordinates.  If the entity's coordinates are 0/0/0 it doesn't send you there.

Reply:
My bad I totally misunderstood what you were doing there, didn't realize you wantes to avoid a load screen for same zone and avoid pos 0/0/0, I sort of skim read the changes coz cell phone I'm sorry